internal/pwru/utils.go

```diff
@@ -15,15 +17,45 @@ import (
 
 type Funcs map[string]int
 
+// getAvailableFilterFunctions return list of functions to which it is possible
+// to attach kprobes.
+func getAvailableFilterFunctions() (map[string]struct{}, error) {
+	availableFuncs := make(map[string]struct{})
+	path := "/sys/kernel/debug/tracing/available_filter_functions"
+
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open %s: %v", path, err)
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		availableFuncs[scanner.Text()] = struct{}{}
+	}
+
```

Comment:
Let's check `scanner.Err()` here:

```suggestion
if err := scanner.Err(); err != nil {
	return nil, err
}
```